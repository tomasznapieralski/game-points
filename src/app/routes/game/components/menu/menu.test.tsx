import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from '@testing-library/react';

import Menu from './menu';

afterEach(cleanup);

test('Snapshot Correct Component', () => {
  const tree = renderer
    .create(
      <Menu
        collectedItems={[]}
        startGameHandler={jest.fn()}
        stopGameHandler={jest.fn()}
        isGameRunning={false}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Table not displayed when empty, scores displayed always', () => {
  const { container } = render(
    <Menu
      collectedItems={[]}
      startGameHandler={jest.fn()}
      stopGameHandler={jest.fn()}
      isGameRunning={false}
    />
  );

  expect(container.getElementsByClassName('scores-table')).toHaveLength(0);
  expect(container.getElementsByClassName('scores-list')).toHaveLength(1);
})

test('Interactions', () => {
  const mockStartGameHandler = jest.fn();
  const mockStopGameHandler = jest.fn();

  let { container } = render(
    <Menu
      collectedItems={[]}
      startGameHandler={mockStartGameHandler}
      stopGameHandler={mockStopGameHandler}
      isGameRunning={false}
    />
  );

  expect(container.getElementsByClassName('menu__footer-button--green')).toHaveLength(1);
  expect(container.getElementsByClassName('menu__footer-button--red')).toHaveLength(0);
  fireEvent.click(container.getElementsByClassName('menu__footer-button')[0]);
  expect(mockStartGameHandler).toBeCalled();
  expect(mockStopGameHandler).not.toBeCalled();

  mockStartGameHandler.mockReset();
  mockStopGameHandler.mockReset();

  container = render(
    <Menu
      collectedItems={[]}
      startGameHandler={mockStartGameHandler}
      stopGameHandler={mockStopGameHandler}
      isGameRunning={true}
    />
  ).container;

  expect(container.getElementsByClassName('menu__footer-button--green')).toHaveLength(0);
  expect(container.getElementsByClassName('menu__footer-button--red')).toHaveLength(1);
  fireEvent.click(container.getElementsByClassName('menu__footer-button')[0]);
  expect(mockStartGameHandler).not.toBeCalled();
  expect(mockStopGameHandler).toBeCalled();
})
