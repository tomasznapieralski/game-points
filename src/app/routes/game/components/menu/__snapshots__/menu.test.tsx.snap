// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Correct Component 1`] = `
<div
  className="menu"
>
  <div
    className="menu__title"
  >
    Player Items
  </div>
  <div
    className="menu__space"
  />
  <div
    className="menu__footer"
  >
    <ul
      className="scores-list"
    >
      <li
        className="scores-list__item"
      >
        <div>
          Base points:
        </div>
        <div>
          0
        </div>
      </li>
      <li
        className="scores-list__item"
      >
        <div>
          Bonus points:
        </div>
        <div>
          0
        </div>
      </li>
      <li
        className="scores-list__item"
      >
        <div>
          Total points:
        </div>
        <div>
          0
        </div>
      </li>
    </ul>
    <button
      className="menu__footer-button menu__footer-button--green"
      onClick={[MockFunction]}
    >
      Start Game
    </button>
  </div>
</div>
`;
